florist/api/db/entities.py

```diff
@@ -67,10 +75,127 @@ class Job(BaseModel):
+
+        :param status: (JobStatus) The status of the jobs to be returned.
+        :param limit: (int) the limit amount of records that should be returned.
+        :param database: (motor.motor_asyncio.AsyncIOMotorDatabase) The database where the job collection is stored.
+        :return: (List[Job]) The list of jobs with the given status int the database.
+        """
+        status = jsonable_encoder(status)
+
+        job_collection = database[JOB_COLLECTION_NAME]
+        result = await job_collection.find({"status": status}).to_list(limit)
+        assert isinstance(result, list)
+        return [Job(**r) for r in result]
+
+    async def create(self, database: AsyncIOMotorDatabase) -> str:
+        """
+        Save this instance under a new record in the database.
+
+        :param database: (motor.motor_asyncio.AsyncIOMotorDatabase) The database where the job collection is stored.
+        :return: (str) the new job record's id.
+        """
+        json_job = jsonable_encoder(self)
+        result = await database[JOB_COLLECTION_NAME].insert_one(json_job)
+        assert isinstance(result.inserted_id, str)
+        return result.inserted_id
+
+    async def set_uuids(self, server_uuid: str, client_uuids: List[str], database: AsyncIOMotorDatabase) -> None:
+        """
+        Save the server and clients' UUIDs in the database under the current job's id.
+
+        :param server_uuid: [str] the server_uuid to be saved in the database.
+        :param client_uuids: List[str] the list of client_uuids to be saved in the database.
+        :param database: (motor.motor_asyncio.AsyncIOMotorDatabase) The database where the job collection is stored.
+        """
+        assert self.clients_info is not None and len(self.clients_info) == len(client_uuids), (
+            "self.clients_info and client_uuids must have the same length "
+            f"({'None' if self.clients_info is None else len(self.clients_info)}<>{len(client_uuids)})."
+        )
+
+        job_collection = database[JOB_COLLECTION_NAME]
+
+        self.server_uuid = server_uuid
+        await job_collection.update_one({"_id": self.id}, {"$set": {"server_uuid": server_uuid}})
+        for i in range(len(client_uuids)):
+            self.clients_info[i].uuid = client_uuids[i]
+            await job_collection.update_one({"_id": self.id}, {"$set": {f"clients_info.{i}.uuid": client_uuids[i]}})
+
+    async def set_status(self, status: JobStatus, database: AsyncIOMotorDatabase) -> None:
+        """
+        Save the status in the database under the current job's id.
+
+        :param status: (JobStatus) the status to be saved in the database.
+        :param database: (motor.motor_asyncio.AsyncIOMotorDatabase) The database where the job collection is stored.
+        """
+        job_collection = database[JOB_COLLECTION_NAME]
+        self.status = status
+        await job_collection.update_one({"_id": self.id}, {"$set": {"status": status.value}})
```

Comment:
For update operations, should we throw an error if the record does not already exist in the db?

Reply:
Good one, will add that

Reply:
Nice!